gateway/src/main/java/org/georchestra/gateway/security/oauth2/OAuth2Configuration.java

```diff
@@ -91,15 +94,41 @@ public ReactiveOAuth2AccessTokenResponseClient<OAuth2AuthorizationCodeGrantReque
         return client;
     }
 
+    /**
+     * Custom JWT decoder factory to use the web client that can be set up to go
+     * through an HTTP proxy
+     */
     @Bean
-    public ReactiveOAuth2UserService<OAuth2UserRequest, OAuth2User> reactiveOAuth2UserService(
+    public ReactiveJwtDecoderFactory<ClientRegistration> idTokenDecoderFactory(
+            @Qualifier("oauth2WebClient") WebClient oauth2WebClient) {
+        return (clientRegistration) -> {
+            ProviderDetails providerDetails = clientRegistration.getProviderDetails();
+            String jwkSetUri = providerDetails.getJwkSetUri();
+            return NimbusReactiveJwtDecoder//
+                    .withJwkSetUri(jwkSetUri)//
+                    .jwsAlgorithm(SignatureAlgorithm.RS256)//
```

Comment:
Wondering if setting an algorithm is necessary, the supported ones are probably indicated in the issuer-uri endpoint, but then which one to prefer ?
I can live with this though

Reply:
I don't know unfortunately...